dtfs-central-api/src/services/utilisation-report-data-validator/index.test.ts

```diff
@@ -0,0 +1,140 @@
+
+      expect(missingHeaderErrors.length).toBe(0);
+    });
+
+    it('returns multiple errors if multiple headers are missing', () => {
+      const csvDataRowWithIncorrectlySpeltFacilityIdAndCurrency = {
+        'bank facility reference': { value: 'Britannia Energy GEF', column: 'A', row: 1 },
+        exporter: { value: 'test exporter', column: 'C', row: 1 },
+        'base curency': { value: 'GBP', column: 'D', row: 1 },
+        'facility limit': { value: '600000', column: 'E', row: 1 },
+        'facility utilisation': { value: '34538e.54', column: 'F', row: 1 },
+        'total fees accrued for the period': { value: '367.23', column: 'G', row: 1 },
+        'fees paid to ukef for the period': { value: '367.23', column: 'H', row: 1 },
+        'payment reference': { value: 'Britannia Energy / 3001175147', column: 'I', row: 1 },
+      };
+
+      const { missingHeaderErrors } = validateUtilisationReportCsvHeaders(csvDataRowWithIncorrectlySpeltFacilityIdAndCurrency);
+
+      expect(missingHeaderErrors.length).toBe(3);
+      expect(missingHeaderErrors[0].errorMessage).toBe('UKEF facility ID header is missing or spelt incorrectly');
+    });
+  });
+
+  describe('validateUtilisationReportCsvCellData', () => {
+    // This test mocks out all the function from utilisation-report-cell-validators.js and
+    // tests that if headers are available then the respective cell validator function is called on that data
+
+    it('calls the generate error functions for headers that are present', () => {
+      const csvData = [
+        {
+          'bank facility reference': { value: 'Britannia Energy GEF', column: 'A', row: 1 },
+          'ukef facility id': { value: '20001371', column: 'B', row: 1 },
+          exporter: { value: 'test exporter', column: 'C', row: 1 },
+          'base currency': { value: 'GBP', column: 'D', row: 1 },
+          'facility limit': { value: '600000', column: 'E', row: 1 },
+          'facility utilisation': { value: '34538e.54', column: 'F', row: 1 },
+          'total fees accrued for the period': { value: '367.23', column: 'G', row: 1 },
+          'fees paid to ukef for the period': { value: '367.23', column: 'H', row: 1 },
+          'payment reference': { value: 'Britannia Energy / 3001175147', column: 'I', row: 1 },
+        },
+      ];
+
+      const availableHeaders = [UTILISATION_REPORT_HEADERS.UKEF_FACILITY_ID, UTILISATION_REPORT_HEADERS.BASE_CURRENCY];
+
+      validateUtilisationReportCsvCellData(csvData, availableHeaders);
+      expect(generateUkefFacilityIdError).toHaveBeenCalledWith({ value: '20001371', column: 'B', row: 1 }, 'test exporter');
+      expect(generateBaseCurrencyError).toHaveBeenCalledWith({ value: 'GBP', column: 'D', row: 1 }, 'test exporter');
+      expect(generateFacilityUtilisationError).not.toHaveBeenCalled();
+    });
+
+    it('calls the generate payment currency and exchange rate error functions even if no headers are present', () => {
```

Comment:
can all these its start with 'should'?

Reply:
This doesn't seem to be a consistent pattern from a search through the codebase, so I can change it but keen to understand if this is something we should be doing going forwards/something all devs are aware of?

Reply:
I believe that this is what we will be doing going forward but can discuss first